adapters/repos/db/inverted/searcher.go

```diff
@@ -323,18 +327,42 @@ func (s *Searcher) extractPropValuePairs(ctx context.Context,
 		eg.Go(func() error {
 			ctx := concurrency.ContextWithFractionalBudget(ctx, concurrencyReductionFactor, concurrency.NUMCPU)
 			child, err := s.extractPropValuePair(ctx, &clause, className)
+			// check for stopword errors on ContainsAny operator only at the end
+			if err != nil && errors.Is(err, ErrOnlyStopwords) && operator == filters.ContainsAny {
+				return nil
+			}
 			if err != nil {
 				return fmt.Errorf("nested clause at pos %d: %w", i, err)
 			}
 			children[i] = child
-
 			return nil
 		}, clause)
 	}
 	if err := eg.Wait(); err != nil {
 		return nil, fmt.Errorf("nested query: %w", err)
 	}
-	return children, nil
+	finalChildren := make([]*propValuePair, 0, len(children))
+	for _, child := range children {
+		if child != nil {
+			finalChildren = append(finalChildren, child)
+		}
+	}
```

Comment:
no need to create new slice, gaps can be removed from original `children` slice and slice shortened at the end.